student-projects/lilach-elayza/module-II/class-023-21-09-2023/home-project/app.ts

```diff
@@ -0,0 +1,240 @@
+type Gender = "Male" | "Female";
+type Race = "Human" | "Elf" | "Dwarf" | "Orc";
+type Class = "Fighter" | "Barbarian" | "Wizard" | "Cleric";
+type AttributePoints = {
+  availablePoints: number;
+  strengh: number;
+  dexterity: number;
+  intelligence: number;
+  wisdom: number;
+  charisma: number;
+};
+type Character = {
+  name: string;
+  gender: Gender;
+  race: Race;
+  class: Class;
+  attributePoints: AttributePoints;
+  image: string;
+};
+type CharacterArray = Character[];
+let characters: CharacterArray;
+
+const retrieveCharactersString = localStorage.getItem("characters");
+if (retrieveCharactersString == null) {
+    characters = [];
+} else {
+    characters = JSON.parse(retrieveCharactersString);
+}
+
+let totalPoints = 55;
```

Comment:
`totalPoints` does not change during the run, it should be a `const` variable.
In the future it should probably be derived from the level (which is derived from the exp).
It's important to maintain single source of truth in cases like these.